share/wake/lib/core/list.wake

```diff
@@ -226,3 +226,16 @@ export def unzip = match _
   Pair a b, t =
     def Pair u v = unzip t
     Pair (a, u) (b, v)
+
+# Group a list by some comparison function.
+# Elements which compare equal are placed into the same bucket.
+# Buckets preserve the order of the original list.
+# The buckets appear in the sort-order specified by cmp.
+# Example: seq 5 | groupBy (_/3 < _/3)
+#          (0,1,2,Nil), (3,4,Nil), Nil
```

Comment:
Can we make examples low-density, and to have explicit arguments (ie, no `_`)
```
def cmp a b = (a/2 < b/3)
def list = seq 10
groupBy cmp list
   <output>
```

Reply:
Sure. How about the new version?

Reply:
I've included two examples now. The modulo one (which I think you were expecting) and the mind bending one.

Reply:
Yeah it looks better